File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
result=[]
while True:
    triangle=list(map(int, input().split()))
    triangle.sort()
    if triangle[2]==0:
        break
    if triangle[2]>=triangle[0]+triangle[1]:
        result.append("Invalid")
    else:
        if triangle[0]==triangle[2]:
            result.append("Equilateral")
        elif triangle[0]==triangle[1] or triangle[1]==triangle[2]:
            result.append("Isosceles")
        else:
            result.append("Scalene")

for elem in result:
    print(elem)

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        while (true) {
            String[] nums = br.readLine().split(" ");

            int a = Integer.parseInt(nums[0]);
            int b = Integer.parseInt(nums[1]);
            int c = Integer.parseInt(nums[2]);

            if (a==0 && b==0 && c==0) {
                break;
            }

            if (a == b && b == c) {
                bw.write("Equilateral\n");
            } else if (a != b && b != c && a != c) {
                int max = Math.max(a,Math.max(b, c));
                if(a+b+c - max > max) {
                    bw.write("Scalene\n");
                } else{bw.write("Invalid\n");}
            } else {
                int max = Math.max(a,Math.max(b, c));
                if(a+b+c - max > max) {
                    bw.write("Isosceles\n");
                } else{bw.write("Invalid\n");}
            }
        }

        bw.flush();
        bw.close();
    }
}

